src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java

```diff
@@ -3858,6 +3858,19 @@ static Response deaccessionDataset(Integer datasetId, String version, String dea
                 .post("/api/datasets/" + datasetId + "/versions/" + version + "/deaccession");
     }
 
+    static Response deaccessionDatasetByGlobalId(String globalId, String version, String deaccessionReason, String deaccessionForwardURL, String apiToken) {
+        JsonObjectBuilder jsonObjectBuilder = Json.createObjectBuilder();
+        jsonObjectBuilder.add("deaccessionReason", deaccessionReason);
+        if (deaccessionForwardURL != null) {
+            jsonObjectBuilder.add("deaccessionForwardURL", deaccessionForwardURL);
+        }
+        String jsonString = jsonObjectBuilder.build().toString();
+        return given()
+                .header(API_TOKEN_HTTP_HEADER, apiToken)
+                .body(jsonString)
+                .post("/api/datasets/:persistentId/versions/" + version + "/deaccession?persistentId=" + globalId);
+    }
+
```

Comment:
I guess it's ok to add a whole new method but if you want to pass a PID (I would suggest this term over GlobalID), you could use this pattern you'll see elsewhere in UtilIT:

```
String idInPath = datasetIdOrPersistentId; // Assume it's a number.
String optionalQueryParam = ""; // If idOrPersistentId is a number we'll just put it in the path.
if (!NumberUtils.isCreatable(datasetIdOrPersistentId)) {
    idInPath = ":persistentId";
    optionalQueryParam = "?persistentId=" + datasetIdOrPersistentId;
}
```

Reply:
I made some changes, please let me know if this is better.

Reply:
It's better, but did you find a datasetIdOrPersistentId example in UtilIT? Do you see what I mean, how we can reduce code duplication?

Reply:
I will change that in a bit 😅

Reply:
Ok, I made some changes, let me know what you think. 😄